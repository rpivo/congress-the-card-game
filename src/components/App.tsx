import React from 'react';
import Card from '@components/Card';

const App = () => <div className='app' style={{ border: '1px solid black' }}><Card /></div>;

export default App;