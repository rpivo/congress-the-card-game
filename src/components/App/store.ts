import React from 'react';
import cardData from '@utilities/cards.json';
import { createStringEnum } from '@utilities/types';

const {
  DRAW_CARD,
  DISCARD_CARD,
  END_TURN,
  HAND_FULL,
  SHOW_HAND,
  HIDE_HAND,
} = Actions;

export type Actions = keyof typeof Actions;

const getCardOrder = (): number[] => {
  const cardCount = Object.keys(cardData).length;
  const cardIDs = Array.from([...Array(cardCount).keys()]);
  for (let i = cardCount - 1; i > 0; i--) {
    const j = Math.floor(Math.random() * (i + 1));
    [cardIDs[i], cardIDs[j]] = [cardIDs[j], cardIDs[i]];
  }
  return cardIDs;
};

export type StateShape = {
  canDrawCard: boolean;
  cardOrder: number[];
  handCards: number[];
  notifyHandIsFull: boolean;
  playAreaCards: number[];
  shouldShowHand: boolean;
};

export const getDefaultState = (): StateShape => {
  const cardOrder = getCardOrder();
  return {
    canDrawCard: true,
    cardOrder,
    handCards: [],
    notifyHandIsFull: false,
    playAreaCards: cardOrder.splice(0, 5),
    shouldShowHand: false,
  };
};

export const Reducer = (state: StateShape, action: Actions): StateShape => {
  switch (action) {
    case 'DRAW_CARD':
      return {
        ...state,
        canDrawCard: false,
        handCards: [
          ...state.handCards,
          state.cardOrder.shift(),
        ] as number[],
        notifyHandIsFull: false,
        shouldShowHand: true,
      };
    case 'END_TURN':
      return {
        ...state,
        canDrawCard: true,
      };
    case 'HAND_FULL':
      return {
        ...state,
        notifyHandIsFull: !state.notifyHandIsFull,
      };
    case 'SHOW_HAND':
      return {
        ...state,
        shouldShowHand: true,
      };
    case 'HIDE_HAND':
    default:
      return {
        ...state,
        shouldShowHand: false,
      };
  }
};

type Context = {
  dispatch: React.Dispatch<Actions>;
  state: StateShape;
};

export const Context = React.createContext({} as Context);
