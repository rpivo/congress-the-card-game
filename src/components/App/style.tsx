import styled from 'styled-components';

const Style = styled.div`
  border: 1px solid red;
  color: #333;
  font-family: Arial;
  font-size: 14px;
  font-weight: bold;
`;

export default Style;
