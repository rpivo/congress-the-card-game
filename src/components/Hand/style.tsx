import styled from 'styled-components';

type HandStyle = {
  cardCount: number;
};

const Style = styled.div<HandStyle>`
  background: rgba(0, 0, 0, 0.8);
  bottom: 0;
  display: grid;
  grid-row-gap: 8px;
  grid-template-columns: ${props => `repeat(${props.cardCount}, 175px [col-start]);`}
  justify-content: center;
  left: 0;
  min-height: 226px;
  padding: 15px;
  position: fixed;
  right: 0;
  transition: bottom 0.1s ease;

  &.hidden {
    bottom: -260px;
  }
`;

export default Style;
