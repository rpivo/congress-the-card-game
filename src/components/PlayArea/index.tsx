import React from 'react';
import Style from './style';
import Card from '@components/Card';

const PlayArea = (): JSX.Element =>
  <Style className='playArea'>
    <Card />
    <Card />
    <Card />
    <Card />
    <Card />
    <Card />
    <Card />
    <Card />
    <Card />
    <Card />
    <Card />
    <Card />
  </Style>;

export default PlayArea;
