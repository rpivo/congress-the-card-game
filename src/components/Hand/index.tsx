import React from 'react';
import { Context } from '@components/App/store';
import { getCards } from '@components/Card/util.tsx';
import { Actions } from '@utilities/enums';
import Style from './style';

type HandProps = {
  activeCard: number;
  handleCardClick: (id: number, isHandCard?: boolean) => void;
  mouseCoordinates: { x: number; y: number };
}

const Hand = ({
  activeCard,
  handleCardClick,
  mouseCoordinates,
}: Readonly<HandProps>): JSX.Element => {
  const { dispatch, state } = React.useContext(Context);
  const { handCards, shouldShowHand } = state;

  const handleXIconClick = (id: number) =>
    dispatch({
      data: { id },
      type: Actions.DISCARD_CARD,
    });

  const cards = getCards({
    activeCard,
    cardIDs: handCards,
    handleMouseDown: handleCardClick,
    handleXIconClick,
    isHandCard: true,
    mouseCoordinates,
  });

  return (
    <Style
      cardCount={handCards.length}
      className={`hand${shouldShowHand ? '' : ' hidden'}`}
      onClick={(event: React.MouseEvent<HTMLInputElement>): void => event.stopPropagation()}>
      {cards.length ? cards : <p className='empty'>Your hand is empty.</p>}
    </Style>
  );
};

export default Hand;
