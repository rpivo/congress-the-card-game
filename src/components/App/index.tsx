import React, { useState } from 'react';
import HandIcon from '@components/HandIcon';
import PlayArea from '@components/PlayArea';
import Style from './style';

const App = (): JSX.Element => {
  const [activeCard, setActiveCard] = useState(-1);
  return (
    <Style className='app' onClick={() => setActiveCard(-1)}>
      <PlayArea activeCard={activeCard} setActiveCard={setActiveCard} />
      <HandIcon />
    </Style>
  );
};

export default App;
