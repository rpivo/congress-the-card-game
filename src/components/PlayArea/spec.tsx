import { mount } from 'enzyme';
import React from 'react';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Card from '@components/Card';
import Deck from '@components/Deck';
import PlayArea from '@components/PlayArea';

describe('PlayArea', () => {
  const setActiveCard = jest.fn();
  const PlayAreaMock = <PlayArea activeCard={''} setActiveCard={setActiveCard} />;

  it('should render correctly', () => {
    const tree = renderer
      .create(PlayAreaMock)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('Cards', () => {
    it('should render the correct amount of Card components', () => {
      const wrapper = mount(PlayAreaMock);
      expect(wrapper.find(Card)).toHaveLength(6);
    });
  });

  describe('Deck', () => {
    it('should render the Deck component', () => {
      const wrapper = mount(PlayAreaMock);
      expect(wrapper.find(Deck)).toHaveLength(1);
    });

    it('should should initially contain the "Draw a Card." paragraph', () => {
      const wrapper = mount(PlayAreaMock);
      expect(wrapper.find(Deck).find('p')).toHaveLength(1);
    });

    it('should not contain a "Draw a card." paragraph tag after being clicked', () => {
      const wrapper = mount(PlayAreaMock);
      const selector = wrapper.find('.stackedCard').at(0).prop('onClick');
      const event = {} as React.MouseEvent;
      if (selector) act(() => selector(event));
      wrapper.update();
      expect(wrapper.find('p')).toHaveLength(0);
    });
  });
});
