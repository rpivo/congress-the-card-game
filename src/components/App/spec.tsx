import { mount } from 'enzyme';
import React from 'react';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import { StopPropagationMouseEvent } from '@utilities/types';
import App from '@components/App';
import Hand from '@components/Hand';
import PlayArea from '@components/PlayArea';

describe('App', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<App />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('activeCard state', () => {
    it('should render the PlayArea', () => {
      const wrapper = mount(<App />);
      expect(wrapper.find(PlayArea)).toHaveLength(1);
    });

    it('should add/remove the .active class from Card components on mousedown', () => {
      const wrapper = mount(<App />);
      wrapper.find('.card').at(0).simulate('mousedown');
      expect(wrapper.find('.card').at(0).hasClass('active')).toBe(true);
      wrapper.find('.card').at(0).simulate('mousedown');
      expect(wrapper.find('.card').at(0).hasClass('active')).toBe(false);
    });

    it('should remove the .active class from Card components when PlayArea is clicked', () => {
      const wrapper = mount(<App />);
      wrapper.find('.card').at(0).simulate('mousedown');
      expect(wrapper.find('.card').at(0).hasClass('active')).toBe(true);
      wrapper.find('.playArea').at(0).simulate('click');
      expect(wrapper.find('.card').at(0).hasClass('active')).toBe(false);
    });
  });

  describe('canDrawCard state', () => {
    it(`should show the "Take a Card" ArrowIcon after the Deck has already been clicked and the
    "End Turn" ArrowIcon is subsequently clicked`, () => {
      const wrapper = mount(<App />);
      expect(wrapper.find('.drawCardIcon').at(0)).toHaveLength(1);
      const selector = wrapper.find('.stackedCard').at(0).prop('onClick');
      const event = {} as React.MouseEvent;
      if (selector) act(() => selector(event));
      wrapper.update();
      expect(wrapper.find('.drawCardIcon').at(0)).toHaveLength(0);
      const endTurnSelector = wrapper.find('.endTurnIcon').at(0).prop('onClick');
      if (endTurnSelector) act(() => endTurnSelector(event));
      wrapper.update();
      expect(wrapper.find('.drawCardIcon').at(0)).toHaveLength(1);
    });

    it(`should perform no action if the "End Turn" ArrowIcon while the "Take a Card" ArrowIcon is
    already showing`, () => {
      const wrapper = mount(<App />);
      expect(wrapper.find('.drawCardIcon').at(0)).toHaveLength(1);
      const event = {} as React.MouseEvent;
      const endTurnSelector = wrapper.find('.endTurnIcon').at(0).prop('onClick');
      if (endTurnSelector) act(() => endTurnSelector(event));
      wrapper.update();
      expect(wrapper.find('.drawCardIcon').at(0)).toHaveLength(1);
    });

    it('should perform no action if an ArrowIcon with class drawCardIcon is clicked', () => {
      const wrapper = mount(<App />);
      expect(wrapper.find('.drawCardIcon').at(0)).toHaveLength(1);
      const event = {} as React.MouseEvent;
      const drawCardSelector = wrapper.find('.drawCardIcon').at(0).prop('onClick');
      if (drawCardSelector) act(() => drawCardSelector(event));
      wrapper.update();
      expect(wrapper.find('.drawCardIcon').at(0)).toHaveLength(1);
    });
  });

  describe('shouldShowHand state', () => {
    it('should not hide the Hand component if it\'s already displaying and it\'s clicked', () => {
      const wrapper = mount(<App />);
      expect(wrapper.find(Hand).find('div').at(0).hasClass('hidden')).toBe(true);
      wrapper.find('.handIcon').at(0).simulate('click');
      expect(wrapper.find(Hand).find('div').at(0).hasClass('hidden')).toBe(false);
      wrapper.find('.hand').at(0).simulate('click');
      expect(wrapper.find(Hand).find('div').at(0).hasClass('hidden')).toBe(false);
    });

    it(`should hide the Hand component if it's already displaying and a Card component is
    clicked`, () => {
      const wrapper = mount(<App />);
      expect(wrapper.find(Hand).find('div').at(0).hasClass('hidden')).toBe(true);
      wrapper.find('.handIcon').at(0).simulate('click');
      expect(wrapper.find(Hand).find('div').at(0).hasClass('hidden')).toBe(false);
      wrapper.find('.card').at(0).simulate('mousedown');
      expect(wrapper.find(Hand).find('div').at(0).hasClass('hidden')).toBe(true);
    });

    it('should not display the drawCardIcon on the Deck once the deck is clicked', () => {
      const wrapper = mount(<App />);
      expect(wrapper.find('.drawCardIcon').at(0)).toHaveLength(1);
      const selector = wrapper.find('.stackedCard').at(0).prop('onClick');
      const event = {} as React.MouseEvent;
      if (selector) act(() => selector(event));
      wrapper.update();
      expect(wrapper.find('.drawCardIcon').at(0)).toHaveLength(0);
    });

    it('should show the Hand if the Deck is clicked when the drawCardIcon is showing', () => {
      const wrapper = mount(<App />);
      expect(wrapper.find('.hand').find('.hidden').at(0)).toHaveLength(1);
      expect(wrapper.find('.drawCardIcon').at(0)).toHaveLength(1);
      const selector = wrapper.find('.stackedCard').at(0).prop('onClick');
      const event = {} as React.MouseEvent;
      if (selector) act(() => selector(event));
      wrapper.update();
      expect(wrapper.find('.hand').find('.hidden').at(0)).toHaveLength(0);
    });

    it(`should not show the Hand if the Deck is clicked when the drawCardIcon is not 
    showing`, () => {
      const wrapper = mount(<App />);
      expect(wrapper.find(Hand).find('div').at(0).hasClass('hidden')).toBe(true);
      const event = {} as React.MouseEvent;
      const deckSelector = wrapper.find('.stackedCard').find('div').at(0).prop('onClick');
      if (deckSelector) act(() => deckSelector(event));
      wrapper.update();
      expect(wrapper.find(Hand).find('div').at(0).hasClass('hidden')).toBe(false);
      const appSelector = wrapper.find('div').at(0).prop('onClick');
      if (appSelector) act(() => appSelector(event));
      wrapper.update();
      expect(wrapper.find(Hand).find('div').at(0).hasClass('hidden')).toBe(true);
    });

    it('should show the hand when the HandIcon is clicked', () => {
      const wrapper = mount(<App />);
      expect(wrapper.find(Hand).find('div').at(0).hasClass('hidden')).toBe(true);
      const event = {
        altKey: false,
        button: 0,
        buttons: 0,
        clientX: 0,
        clientY: 0,
        ctrlKey: false,
        metaKey: false,
        movementX: 0,
        movementY: 0,
        pageX: 0,
        pageY: 0,
        relatedTarget: null,
        screenX: 0,
        screenY: 0,
        shiftKey: false,
        stopPropagation: jest.fn(),
      } as unknown as StopPropagationMouseEvent;
      const selector = wrapper.find('.handIcon').find('div').at(0).prop('onClick');
      if (selector) act(() => selector(event));
      wrapper.update();
      expect(wrapper.find(Hand).find('div').at(0).hasClass('hidden')).toBe(false);
      const updatedSelector = wrapper.find('.handIcon').find('div').at(0).prop('onClick');
      if (updatedSelector) act(() => updatedSelector(event));
      wrapper.update();
      expect(wrapper.find(Hand).find('div').at(0).hasClass('hidden')).toBe(false);
    });
  });

  describe('handleClick', () => {
    it(`should remove active card state when an active card exists and the App component is
    clicked`, () => {
      const wrapper = mount(<App />);
      wrapper.find('.card').at(0).simulate('mousedown');
      wrapper.find(App).simulate('click');
      expect(wrapper.find('.active')).toHaveLength(0);
    });

    it(`should hide Hand component when Hand component is showing and the App component is
    clicked`, () => {
      const wrapper = mount(<App />);
      expect(wrapper.find(Hand).find('div').at(0).hasClass('hidden')).toBe(true);
      wrapper.find('.handIcon').at(0).simulate('click');
      expect(wrapper.find(Hand).find('div').at(0).hasClass('hidden')).toBe(false);
      wrapper.find(App).simulate('click');
      expect(wrapper.find(Hand).find('div').at(0).hasClass('hidden')).toBe(true);
    });
  });
});
