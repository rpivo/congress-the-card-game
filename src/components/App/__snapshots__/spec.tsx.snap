// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div
  className="sc-AxgMl grnyfY app"
  onClick={[Function]}
>
  <div
    className="sc-AxhUy isiHme playArea"
  >
    <div
      className="sc-AxirZ dLSlov card"
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <p
        className="title"
      >
        Alabama
      </p>
      <p
        className="subtext subtitle"
      >
        state
      </p>
    </div>
    <div
      className="sc-AxirZ dLSlov card"
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <p
        className="title"
      >
        Alaska
      </p>
      <p
        className="subtext subtitle"
      >
        state
      </p>
    </div>
    <div
      className="sc-AxirZ dLSlov card"
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <p
        className="title"
      >
        Arizona
      </p>
      <p
        className="subtext subtitle"
      >
        state
      </p>
    </div>
    <div
      className="sc-AxirZ dLSlov card"
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <p
        className="title"
      >
        Arkansas
      </p>
      <p
        className="subtext subtitle"
      >
        state
      </p>
    </div>
    <div
      className="sc-AxirZ dLSlov card"
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <p
        className="title"
      >
        California
      </p>
      <p
        className="subtext subtitle"
      >
        state
      </p>
    </div>
    <div
      className="sc-AxirZ dLSlov card"
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <p
        className="title"
      >
        Colorado
      </p>
      <p
        className="subtext subtitle"
      >
        state
      </p>
    </div>
    <div
      className="sc-AxirZ dLSlov card"
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <p
        className="title"
      >
        Connecticut
      </p>
      <p
        className="subtext subtitle"
      >
        state
      </p>
    </div>
    <div
      className="sc-AxirZ dLSlov card"
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <p
        className="title"
      >
        Delaware
      </p>
      <p
        className="subtext subtitle"
      >
        state
      </p>
    </div>
    <div
      className="sc-AxirZ dLSlov card"
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <p
        className="title"
      >
        Sonny Perdue
      </p>
      <p
        className="subtext subtitle"
      >
        cabinet
      </p>
      <p
        className="subtext ability1Title"
      >
        Judo Chop
         - 
        Deal 20 damage to player.
      </p>
    </div>
    <div
      className="sc-AxirZ dLSlov card"
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <p
        className="title"
      >
        William Barr
      </p>
      <p
        className="subtext subtitle"
      >
        cabinet
      </p>
      <p
        className="subtext ability1Title"
      >
        Banshee Call
         - 
        Skip opponent's next turn.
      </p>
    </div>
    <div
      className="sc-AxirZ dLSlov card"
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <p
        className="title"
      >
        Gina Haspel
      </p>
      <p
        className="subtext subtitle"
      >
        cabinet
      </p>
    </div>
    <div
      className="sc-AxirZ dLSlov card"
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <p
        className="title"
      >
        Lamar Alexander
      </p>
      <p
        className="subtext subtitle"
      >
        congress
      </p>
    </div>
    <div
      className="sc-AxirZ dLSlov deck"
    >
      <div
        className="sc-AxirZ dLSlov stackedCard canDrawCard"
        onClick={[Function]}
      >
        <div
          className="sc-AxjAm lkoMCz drawCardIcon"
        >
          <svg
            height="50"
            viewBox="0 0 703 703"
            width="50"
          >
            <g
              transform="translate(-398 -203)"
            >
              <circle
                cx="351.5"
                cy="351.5"
                fill="#707070"
                r="351.5"
                transform="translate(398 203)"
              />
              <path
                d="M136,136,0,272ZM0,0,136,136Z"
                fill="none"
                stroke="#EAEAEA"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="100"
                transform="translate(805.5 418.5)"
              />
              <line
                fill="none"
                stroke="#EAEAEA"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="100"
                transform="translate(556.5 554.5)"
                x2="385"
              />
            </g>
          </svg>
          <p>
            Take a Card
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-AxhCb cFIxOX handIcon"
    onClick={[Function]}
  >
    <svg
      height="421.425"
      viewBox="0 0 445.805 421.425"
      width="445.805"
    >
      <g
        transform="translate(-195.558 -456.564)"
      >
        <g
          className="svgCard"
          fill="#707070"
          stroke="#fff"
          strokeWidth="12"
          transform="matrix(0.995, 0.105, -0.105, 0.995, 392.732, 495.977)"
        >
          <rect
            height="350"
            rx="15"
            stroke="none"
            width="250"
          />
          <rect
            fill="none"
            height="338"
            rx="9"
            width="238"
            x="6"
            y="6"
          />
        </g>
        <g
          className="svgCard"
          fill="#707070"
          stroke="#fff"
          strokeWidth="12"
          transform="translate(290.401 516.605) rotate(-9)"
        >
          <rect
            height="350"
            rx="15"
            stroke="none"
            width="250"
          />
          <rect
            fill="none"
            height="338"
            rx="9"
            width="238"
            x="6"
            y="6"
          />
        </g>
        <g
          className="svgCard"
          fill="#707070"
          stroke="#fff"
          strokeWidth="12"
          transform="matrix(0.914, -0.407, 0.407, 0.914, 195.558, 558.248)"
        >
          <rect
            height="350"
            rx="15"
            stroke="none"
            width="250"
          />
          <rect
            fill="none"
            height="338"
            rx="9"
            width="238"
            x="6"
            y="6"
          />
        </g>
      </g>
    </svg>
  </div>
  <div
    className="sc-AxiKw fQdyLb hand hidden"
    onClick={[Function]}
  >
    <div
      className="sc-AxirZ dLSlov card"
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <p
        className="title"
      />
      <p
        className="subtext subtitle"
      />
    </div>
    <div
      className="sc-AxirZ dLSlov card"
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <p
        className="title"
      />
      <p
        className="subtext subtitle"
      />
    </div>
    <div
      className="sc-AxirZ dLSlov card"
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <p
        className="title"
      />
      <p
        className="subtext subtitle"
      />
    </div>
  </div>
  <div
    className="sc-AxjAm lkoMCz endTurnIcon"
  >
    <svg
      height="75"
      viewBox="0 0 703 703"
      width="75"
    >
      <g
        transform="translate(-398 -203)"
      >
        <path
          d="M136,136,0,272ZM0,0,136,136Z"
          fill="none"
          stroke="#D0D0D0"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="100"
          transform="translate(805.5 418.5)"
        />
        <line
          fill="none"
          stroke="#D0D0D0"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="100"
          transform="translate(556.5 554.5)"
          x2="385"
        />
      </g>
    </svg>
  </div>
</div>
`;
